File: quick.py
def swap(a, i, j):
    # Function to swap two elements in the array
    a[i], a[j] = a[j], a[i]

def partition(a, lb, ub):
    l = lb
    u = ub
    pivot = a[lb]

    while l < u:
        while l < ub and a[l] <= pivot:
            l += 1
        while a[u] > pivot:
            u -= 1
        if l < u:
            swap(a, l, u)

    swap(a, lb, u)
    return u

def quicksort(a, lb, ub):
    if lb < ub:
        loc = partition(a, lb, ub)
        quicksort(a, lb, loc - 1)
        quicksort(a, loc + 1, ub)

if __name__ == "__main__":
    a = [4, 7, 8, 0, 2, 1]
    lb = 0
    ub = len(a) - 1

    quicksort(a, lb, ub)

    print("Sorted array:")
    for element in a:
        print(element, end=" ")

File: quick.c
#include<stdio.h>

void swap1(int *a,int *b)
 { 
   int temp=*a;
   *a=*b;
   *b=temp;
 
 }

int sort(int a[],int lb,int ub)
  {
    int l=lb,u=ub;
    int p=a[lb];
   while(l<u)
      {
        while(a[l]<=p && l<ub)
          {
            l++;
          }
           while(a[u]>p)
               {
                 u--;
               }
           if(l<u)
            {
               swap1(&a[l],&a[u]);
            }
       }
     swap1(&a[lb],&a[u]);

      return u;
}


void quicksort(int a[],int lb,int ub)
{
      if(lb<ub)
        {
          int loc=sort(a,lb,ub);
          quicksort(a,lb,loc-1);
          quicksort(a,loc+1,ub);
        }
}



void main()
{
   int a[]={4,7,8,0,2,1};
   int lb=0,ub=sizeof(a)/sizeof(a[0])-1;

     quicksort(a,lb,ub);
   printf("sorted array\n");
      for(int i=0;i<6;i++)
        {
           printf("%2d",a[i]);
        }

}